lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -345,14 +350,17 @@ def create_instance(host, ami_spec, subnet_id)
         :instance_type => amisize,
         :disable_api_termination => false,
         :instance_initiated_shutdown_behavior => "terminate",
-        :network_interfaces => [{
+      }
+      unless assoc_pub_ip_addr.nil?
```

Comment:
Favor if modifier over unless when having multiple branches. Could change the unless to `if assoc_pub_ip_addr.present?`